### DISCUSSION TITLE
Error Build libmultihash

### DISCUSSION THREAD

**Discussion by belmix** [coinfoundry/miningcore#969] [Q&A]
Error Build libmultihash in VS2019

**_src\Native\libmultihash\equi\crypto\equihash.h(21,10): fatal error C1083: Can't open file include: boost/static_assert.hpp: No such file or directory,_**

_**src\Native\libmultihash\equi\serialize.h(29,10): fatal error C1083:  Can't open file include: boost/array.hpp: No such file or directory**_

**Comment by oliverw** [comment]
Building the native libraries from source on Windows isn't really supported because setting up the build environment is tricky as you have discovered. If you still want to go that route, the first fix would be to have a look at [this](https://github.com/coinfoundry/miningcore/blob/1992e6c9e146a723b61af74723bb4f125e59d5a8/src/Native/libmultihash/libmultihash.vcxproj#L74) for what's expected to be present.

**Comment by belmix** [comment]
under Linux, there are no instructions for building libraries either

**Reply by oliverw** [reply]
> under Linux, there are no instructions for building libraries either

That's because under Linux, the native libraries get built on the fly while building the .NET project using `dotnet publish -c Release --framework net5.0  -o ../../build`
